app/src/main/java/com/example/myapplication/ui/LoginFragment.kt

```diff
@@ -31,18 +39,28 @@ class LoginFragment : Fragment() {
                 val passwordText = lastNameEdit.text.toString()
                 loginViewModel.fieldsValuesValidation(emailText, passwordText)
             }
+            signUpButton.setOnClickListener {
+                loginViewModel.signUpClicked()
+            }
+            termsAndConditions.setOnClickListener {
+                loginViewModel.termsAndConditionsClicked()
+            }
```

Comment:
There's no need to call the viewmodel here. You can just make the redirection like:
```
signUpButton.setOnClickListener {
    goToSignUp()
}
```
As a general rule, don't call the viewmodel unless you are making a request or something like that.